chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -95,6 +106,27 @@ private[chisel3] trait HasId {
   private[chisel3] def setRef(parent: HasId, index: Int): Unit = setRef(Index(Node(parent), ILit(index)))
   private[chisel3] def setRef(parent: HasId, index: UInt): Unit = setRef(Index(Node(parent), index.ref))
   private[chisel3] def getRef: Arg = _ref.get
+
+  // Implementation of public methods.
+  def signalName = _parent match {
+    case None => getRef.name
```

Comment:
Is there ever a case where a signal wouldn't have a parent (other than because of bugs)? Should it assert out rather than trying to make something up?

I also assume that this is trying to be consistent with how CHIRRTL names are generated. If so, it would be nice to just reference the CHIRRTL generated name, as opposed to possibly duplicating logic here (like concatenating path and signal name in pathName implementation).